src/xo-app/dashboard/stats/index.js

```diff
@@ -0,0 +1,402 @@
+  if (layers[metricKey] === undefined) {
+    layers[metricKey] = 0
+  }
+
+  return ++layers[metricKey]
+}
+
+const getMetricValues = (metrics, metricKey) => (
+  (metrics[metricKey] && metrics[metricKey].values) ||
+  []  // Already fed or not.
+)
+
+const computeStatsAverage = (stats, {
+  layers,
+  mapValue = value => value,
+  metricKey,
+  metrics,
+  timestampStart,
+  valuesRenderer
+}) => {
+  if (!stats) {
+    return
+  }
+
+  const layer = updateLayers(layers, metricKey)
+  const values = getMetricValues(metrics, metricKey)
+
+  forEach(stats, (value, key) => {
+    value = +value
+
+    if (values[key] === undefined) {
+      values.push({
+        value: mapValue(value),
+        date: timestampStart + 3600000 * key
+      })
+    } else {
+      values[key].value = (values[key].value * (layer - 1) + value) / layer
+    }
+  })
+
+  metrics[metricKey] = {
+    key: metricKey,
+    values,
+    renderer: valuesRenderer
+  }
+}
+
+// ===================================================================
+
+const computeCpusAverage = (cpus, params) => {
+  forEach(cpus, (cpu, index) => {
+    computeStatsAverage(cpu, {
+      metricKey: `CPU ${index}`,
+      ...params
+    })
+  })
+
+  const nCpus = cpus.length
+
+  if (!nCpus) {
+    return
+  }
+
+  const { metrics } = params
+  const cpusAvg = cloneDeep(metrics['CPU 0'].values)
+
+  for (let i = 1; i < nCpus; i++) {
+    forEach(metrics[`CPU ${i}`].values, (value, index) => {
+      cpusAvg[index].value += value.value
+    })
+  }
+
+  forEach(cpusAvg, value => { value.value /= nCpus })
+
+  const allCpusKey = 'All CPUs'
+  metrics[allCpusKey] = {
+    key: allCpusKey,
+    values: cpusAvg
+  }
+}
+
+const computeVifsAverage = (vifs, params) => {
+  forEach(vifs, (vifs, vifsType) => {
+    const rw = (vifsType === 'rx') ? 'out' : 'in'
+
+    forEach(vifs, (vif, index) => {
+      computeStatsAverage(vif, {
+        metricKey: `Network ${index} ${rw}`,
+        valuesRenderer: formatSize,
+        ...params
+      })
+    })
+  })
+}
+
+const computePifsAverage = (pifs, params) => {
+  forEach(pifs, (pifs, pifsType) => {
+    const rw = (pifsType === 'rx') ? 'out' : 'in'
+
+    forEach(pifs, (pif, index) => {
+      computeStatsAverage(pif, {
+        metricKey: `NIC ${index} ${rw}`,
+        valuesRenderer: formatSize,
+        ...params
+      })
+    })
+  })
+}
+
+const computeXvdsAverage = (xvds, params) => {
+  forEach(xvds, (xvds, xvdsType) => {
+    const rw = (xvdsType === 'r') ? 'read' : 'write'
+
+    forEach(xvds, (xvd, index) => {
+      computeStatsAverage(xvd, {
+        metricKey: `Disk ${index} ${rw}`,
+        valuesRenderer: formatSize,
+        ...params
+      })
+    })
+  })
+}
+
+const computeLoadAverage = (load, params) => {
+  computeStatsAverage(load, {
+    metricKey: 'Load',
+    ...params
+  })
+}
+
+const computeMemoryUsedAverage = (memoryUsed, {
+  object,
+  ...params
+}) => {
+  const factor = object.type === 'host' ? 1024 : 1
```

Comment:
I want a clear comment explaining why this is necessary!

And it should probably be done in other stats (sparklines, line charts).